ibis/backends/snowflake/__init__.py

```diff
@@ -317,12 +317,15 @@ def from_snowpark(cls, session, *, create_object_udfs: bool = True) -> Backend:
         │ ansonca01 │    16 │
         └───────────┴───────┘
         """
+        import snowflake.connector
+
         backend = cls(_from_snowpark=True)
-        backend._setup_session(
-            con=session._conn._conn,
-            session_parameters={},
-            create_object_udfs=create_object_udfs,
-        )
+        backend.con = session._conn._conn
+        with contextlib.suppress(snowflake.connector.errors.ProgrammingError):
```

Comment:
Looks like this is related to caller's rights versus owner's rights (https://community.snowflake.com/s/question/0D5VI000007waK90AI/alter-session-not-works-inside-snowpark-sessionsql).

We have to handle both cases since it's not within Ibis's control, so this code has to stick around for now.